diff -up stunnel-4.56/src/client.c.log-version stunnel-4.56/src/client.c
--- stunnel-4.56/src/client.c.log-version	2013-03-14 23:54:24.000000000 +0100
+++ stunnel-4.56/src/client.c	2016-03-31 17:17:01.438314029 +0200
@@ -928,6 +928,7 @@ static void print_cipher(CLI *c) { /* pr
 
     if(global_options.debug_level<LOG_INFO) /* performance optimization */
         return;
+    s_log(LOG_INFO, "Negotiated protocol version: %s", SSL_get_version(c->ssl));
     cipher=(SSL_CIPHER *)SSL_get_current_cipher(c->ssl);
     s_log(LOG_INFO, "Negotiated %s ciphersuite: %s (%d-bit encryption)",
         SSL_CIPHER_get_version(cipher), SSL_CIPHER_get_name(cipher),
